src/Controls/src/Core/BindableObject.cs

```diff
@@ -645,12 +652,27 @@ void SetValueActual(BindableProperty property, BindablePropertyContext context,
 					_applying = false;
 				}
 
-				OnPropertyChanged(property.PropertyName);
-
-				property.PropertyChanged?.Invoke(this, original, value);
+				FirePropertyChanged(property, original, value);
 			}
+			else
+			{
+				UpdateHandlerValue(property.PropertyName, !sameValue);
+			}
+		}
+
+		void FirePropertyChanged(BindableProperty property, object original, object value)
```

Comment:
Not ordering the UpdateValue calls like this was an early misfire on our part with MAUI
https://github.com/dotnet/maui/issues/5587

Making the propertychanged handler on the BP fire after the handler fires has led to a handful of hard to track errors